bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/types/XSTime.java

```diff
@@ -119,6 +119,9 @@ public static CalendarType parse_time(String str) {
 		if (dt == null)
 			return null;
 
+		// Adjust for 24:00 (xs:time only) by setting it back to 00:00 without rolling the date forward
+		dt.calendar().set(Calendar.DAY_OF_MONTH, 29);
+
```

Comment:
So if it happens to be that the time passed is 24:00, that would cause the date to roll over...so to prevent that from happening you always make sure you roll the date back?  Doesn't this point to an issue in parseDateTime?